health-services/individual/src/main/java/org/egov/individual/service/IndividualService.java

```diff
@@ -0,0 +1,111 @@
+        final String tenantId = getTenantId(request.getIndividual());
+        List<String> indIdList = idGenService.getIdList(request.getRequestInfo(),
+                tenantId, "individual.id",
+                null, request.getIndividual().size());
+        enrichForCreate(request.getIndividual(), indIdList, request.getRequestInfo());
+        List<Address> addresses = collectFromList(request.getIndividual(),
+                Individual::getAddress);
+        if (!addresses.isEmpty()) {
+            List<String> addressIdList = uuidSupplier().apply(addresses.size());
+            enrichForCreate(addresses, addressIdList, request.getRequestInfo());
+            enrichIndividualIdInAddress(request);
+        }
+        request.setIndividual(request.getIndividual().stream()
+                .map(IndividualService::enrichWithSystemGeneratedIdentifier)
+                        .map(IndividualService::enrichIndividualIdInIdentifiers)
+                .collect(Collectors.toList()));
+        List<Identifier> identifiers = collectFromList(request.getIndividual(),
+                Individual::getIdentifiers);
+        List<String> identifierIdList = idGenService.getIdList(request.getRequestInfo(),
+                tenantId, "identifier.id",
+                null, identifiers.size());
+        enrichForCreate(identifiers, identifierIdList, request.getRequestInfo());
+        individualRepository.save(request.getIndividual(), "save-individual-topic");
+        return request.getIndividual();
+    }
+
+    private static Individual enrichIndividualIdInIdentifiers(Individual individual) {
+        List<Identifier> identifiers = individual.getIdentifiers();
+        identifiers.forEach(identifier -> identifier.setIndividualId(individual.getId()));
+        individual.setIdentifiers(identifiers);
+        return individual;
+    }
+
+    private static void enrichIndividualIdInAddress(IndividualRequest request) {
+        request.getIndividual().stream().filter(individual -> individual.getAddress() != null)
+                .forEach(individual -> individual.getAddress()
+                        .forEach(address -> address.setIndividualId(individual.getId())));
+    }
+
+    private static Function<Integer, List<String>> uuidSupplier() {
+        return integer ->  {
+            List<String> uuidList = new ArrayList<>();
+            for (int i = 0; i < integer; i++) {
+                uuidList.add(UUID.randomUUID().toString());
+            }
+            return uuidList;
+        };
+    }
+
+    private static Individual enrichWithSystemGeneratedIdentifier(Individual individual) {
+        if (individual.getIdentifiers() == null || individual.getIdentifiers().isEmpty()) {
+            List<Identifier> identifiers = new ArrayList<>();
+            identifiers.add(Identifier.builder()
+                    .identifierType("SYSTEM_GENERATED")
+                    .identifierId(UUID.randomUUID().toString())
```

Comment:
Let's use IDGen for system generated registry identifiers

Reply:
Done.